### PR TITLE
Offers questions about other expenses to the user for graph visuals - #397

### PR DESCRIPTION
**Pull request by knod**
Take a look at the preview in the 'Expenses' section. Also, put in some rent or mortgage and see how that shows up in the questions. It's a bit long, but it's based off of the assessment worksheet Project Hope uses. See the last page at https://drive.google.com/drive/folders/0B61uzoaPZg7qUXJuRFhndTFkaTA.

This section of questions is a bit long, but I'm not sure how we could usefully decrease its length.

This doesn't implement any of the visuals or anything, which is why I'm pulling to a feature branch I made ('expenses') off of 'dev'.

It may be worth implementing `<ShowOnYes>` regardless of how this feature. I have a feeling, though, that there's some kind of better abstraction. A `<ShowWhen>` that's more general? With a question component and a content component? The question could trigger showing the content...

### CODE REVIEW

**Comment by knod** [comment]
Fixes that and snapshots.

**Comment by knod** [comment]
Also, `<RenderIfTrue>` is in there too, but it's not finalized and it's not in an appropriate file. How that component's going to be implemented hasn't been worked out yet, but I wanted to at least field test it. You'll also notice that there are ternary operators in there too for simple string choices. It's all a grand experiment!

File: src/forms/ShowOnYes.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ * @param {object} props.children - Components to be revealed.
+ * 
+ * @returns {object} Component
+ */
+class ShowOnYes extends React.Component {
+  constructor (props) {
+    super(props);
+    
+    const show = props.clientPartial[ props.propName ];
+    this.state = { showChildren: show };
```
Comment (wacii) [inline src/forms/ShowOnYes.js]: There shouldn't be a need to use state here. Just use `client[propName]` directly.

File: src/forms/cashflow.js [hunk 2]
```diff
@@ -103,7 +103,37 @@ const CashFlowRow = function ({ generic, timeState, updateClientValue, children
...
+      <div className = { colClassName + ` ` + generic + ` output-number` } >
+        { weekly }
+      </div>
+      <div className = { colClassName + ` ` + generic + ` output-number` } >
+        { monthly }
+      </div>
+      <div className = { colClassName + ` ` + generic + ` output-number` } >
+        { yearly }
+      </div>
+      <div className = { colClassName + ` cashflow-column-last-child` }>
```
Comment (wacii) [inline src/forms/cashflow.js]: It's a little weird to be using template strings here but not interpolation. Not something that needs to be fixed though.

**Comment by knod** [comment]
Oops, never merged this. Will go ahead and do that, since this is on a separate branch from `dev`.

**Pull request merged**